### ISSUE TITLE
Default settings don't work as documented

### ISSUE THREAD

**Issue by fletchgqc** [codecentric/chaos-monkey-spring-boot#70]
To reproduce:
- Add chaos-monkey-spring-boot **v2.0.2** dependency to maven
- Add following application properties:
  ```
  spring.profiles.active=chaos-monkey
  chaos.monkey.enabled=true
  chaos.monkey.assaults.level=1
  ```
- Start application and do something which uses a service.
    - Expected result: action successful with latency
    - Actual result: normal functionality
- Add following application properties:
```
chaos.monkey.watcher.service=true
```
- Start application and do something which uses a service.
    - Expected result: action successful with latency
    - Actual result: action successful with latency

**Comment by fletchgqc** [comment]
I have located the bug. It is here:

```
Class AttackServiceCondition

public boolean matches(ConditionContext context, AnnotatedTypeMetadata metadata) {
    return context.getEnvironment()
            .getProperty("chaos.monkey.watcher.service", "false")
            .matches("(?i:.*true*)");
```
This is a Spring Condition which is evaluated in `ChaosMonkeyConfiguration` during startup to decide whether the `SpringServiceAspect` bean should be created or not. If the bean is not created, no attacks on services can happen until application restart.

As you can see, the default value is false.

The configuration for this property is available to be read from the class `WatcherProperties`:

```
@Value("${service:true}")
private boolean service;
```

Here it is defined as defaulting to true. However, it is never read from here in production code.

Since the property cannot be changed during runtime, it is only read once at startup, in the first block of code above. The obvious solution is for `AttackServiceCondition` to read the value from the `WatcherProperties` bean, instead of directly from the spring Environment. However, this is not possible, because `Conditions` do not have access to beans (see: https://stackoverflow.com/a/52072532/159178).

There are three possible solutions:
1. Change AttackServiceCondition so that it defaults to true when reading the property. 
1. Change the documentation and say that by default, no watchers are enabled. (Arguably, this could be combined with defaulting the property chaos.monkey.enabled to true. If no watchers are enabled by default, why add complication by disabling chaos monkey too? I think that needing to specify the profile is enough "safety".)
1. Remove the condition. Always create the bean. Check the property inside the method `SpringServiceAspect#intercept`.
    - Advantage: property becomes configurable at runtime.
    - Disadvantage: performance implications of this method being called on every call, even if this watcher is disabled (currently it’s only called if the watcher is enabled).

If we go for 1 or 2, I would suggest also removing this and any other unread properties from WatcherProperties, they are misleading.

@MrBW what is your preference? I think I am leaning towards 2. Happy to implement something.

**Comment by MrBW** [comment]
I would like to make the switch and initialize the watcher beans as soon as the Spring Boot App is started with the Chaos Monkey profile "chaos-monkey".

It would give us a lot more options to turn the watcher on and off during runtime.

**Comment by maiksensi** [comment]
@fletchgqcb Can this be put to ready-for-release? 
Is https://github.com/codecentric/chaos-monkey-spring-boot/pull/84 the corresponding MR?

**Comment by fletchgqc** [comment]
Yes. 
Yes.

**Comment by maiksensi** [comment]
Closed by #84
